### ISSUE TITLE
Create rule XXX: Exported symbols should be used

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4416]
#### Detection

- Detect any kind of symbol exported from a file but not imported anywhere within the codebase
- The API of TypeScript compiler should probably have some helper allowing to track references from a symbol across a project.
- Heuristic-like adjustments might needed depending on the findings we get on ruling, especially for library-like projects. For instance, we should not flag library entry points.

#### Noncompliant

```js
// foo.js
export const x = 42;
export const y = 24; // Noncompliant: 'y' never used
```

```js
// bar.js
import { x } from './foo';
```

#### Compliant

```js
import { x, y } from './foo';
```

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-66
